src/python/zquantum/qcbm/target_thermal_states.py

```diff
@@ -0,0 +1,235 @@
+    """Generates random h, J, and where h and J are arrays of random coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd.
+    For reproducibilty, fix random generator seed in the higher level from which this function is called.
+    Args:
+        n_spins: positive number of spins in the Ising system.
+    Returns:
+       external_fields: n_spin coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd
+       two_body_couplings: n_spin x n_spin array of coefficients sampled from a normal distribution with zero mean and sqrt(n_spins) sd
+    """
+    external_fields = np.zeros((n_spins))
+    two_body_couplings = np.zeros((n_spins, n_spins))
+    for i in range(n_spins):
+        external_fields[i] = np.random.normal(0, np.sqrt(n_spins))
+        for j in range(i, n_spins):
+            if i == j:
+                continue
+            two_body_couplings[i, j] = two_body_couplings[j, i] = np.random.normal(
+                0, np.sqrt(n_spins)
+            )
+    return external_fields, two_body_couplings
+
+
+def get_thermal_target_distribution_dict(
+    n_spins: int,
+    temperature: float,
+    hamiltonian_parameters: typing.List[np.array],  # TODO: add docstring
+) -> typing.Dict:
+    """Generates thermal states target distribution, saved in a dict where keys are bitstrings and
+    values are corresponding probabilities according to the Boltzmann Distribution formula.
+
+    Args:
+        n_spins: positive number of spins in the Ising system
+        temperature: temperature factor in the boltzman distribution
+        hamiltonian_parameters: values of hamiltonian parameters, namely external fields and two body couplings.
+
+    Returns:
+       Thermal target distribution.
+       Number of positive spins in the spin state. 
+    """
+    partition_function = 0
+    external_fields, two_body_couplings = hamiltonian_parameters
+    beta = 1.0 / temperature
+    distribution = {}
+
+    for spin in range(int(2 ** n_spins)):
+        ising_bitstring = int2ising(spin, n_spins)
+        energy = 0
+        for i in range(n_spins):
+            energy -= ising_bitstring[i] * external_fields[i]
+            for j in range(n_spins):
+                if j == i + 1:
+                    energy -= (
+                        ising_bitstring[i]
+                        * ising_bitstring[j]
+                        * two_body_couplings[i, j]
+                    )
+                else:
+                    continue
+        boltzmann_factor = np.exp(energy * beta)
+        partition_function += boltzmann_factor
+
+        binary_bitstring = convert_tuples_to_bitstrings([dec2bin(spin, n_spins)])[-1]
+        distribution[binary_bitstring] = boltzmann_factor
+
+    normalized_distribution = {
+        key: value / partition_function for key, value in distribution.items()
+    }
+
+    return normalized_distribution
+
+
+def get_target_bitstring_distribution(
```

Comment:
Going off previous comment, I think this should function should be combined with `get_thermal_target_distribution_dict` as you shouldn't need separate functions.

Reply:
Alternatively, you could make `get_thermal_target_distribution_dict` private, but I think it would make more sense to just build the object as a `BitstringDistribution` directly.

Reply:
Agreed. Done!